lib/rust/parser/src/lexer.rs

```diff
@@ -830,17 +835,21 @@ fn analyze_operator(token: &str) -> token::OperatorProperties {
             }
         }
     }
-    let binary = match precedence_char.unwrap() {
-        '!' => 10,
-        '|' => 11,
-        '^' => 12,
-        '&' => 13,
-        '<' | '>' => 14,
-        '+' | '-' => 15,
-        '*' | '/' | '%' => 16,
-        _ => 17,
+    let (value_op, binary) = match precedence_char.unwrap() {
+        '!' => (true, 10),
+        '|' => (true, 11),
+        '&' => (true, 13),
+        '<' | '>' => (true, 14),
+        '+' | '-' => (true, 15),
+        '*' | '/' | '%' => (true, 16),
+        '^' => (true, 17),
+        _ => (true, 18),
```

Comment:
value_op is always true. Do we expect to extend it in the future? If yes, add a comment.